File: backend/app/schemas/document.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel,ConfigDict
from app.models.document import DocumentStatus
class DocumentOut(BaseModel):
 id:UUID;original_filename:str;display_name:str;content_type:str;file_size:int;status:DocumentStatus;processing_error:str|None;chunk_count:int;created_at:datetime;updated_at:datetime
 model_config=ConfigDict(from_attributes=True)
class DocumentList(BaseModel):items:list[DocumentOut];total:int;page:int;page_size:int
class SourceOut(BaseModel):
 document_id:UUID|None;document_chunk_id:UUID|None;document_name:str;page:int|None;rank:int;score:float
 model_config=ConfigDict(from_attributes=True)

File: frontend/src/types/document.ts
export type Document = {
  id: string;
  original_filename: string;
  display_name: string;
  content_type: string;
  file_size: number;
  status: "uploaded" | "processing" | "ready" | "failed";
  processing_error: string | null;
  chunk_count: number;
  created_at: string;
  updated_at: string;
};
export type DocumentList = {
  items: Document[];
  total: number;
  page: number;
  page_size: number;
};
